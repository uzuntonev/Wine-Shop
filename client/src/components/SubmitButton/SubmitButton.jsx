import React from 'react';
import { Button } from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles((theme) => ({
  submit: {
    margin: theme.spacing(3, 0, 2),
  },
}));
const SubmitButton = ({ title, disabled }) => {
  const classes = useStyles();
  return (
    <Button
      type="submit"
      fullWidth
      variant="contained"
      color="primary"
      className={classes.submit}
      disabled={disabled}
    >
      {title}
    </Button>
  );
};

export default SubmitButton;
