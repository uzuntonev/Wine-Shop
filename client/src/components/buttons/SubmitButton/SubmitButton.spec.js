import React from 'react';
import SubmitButton from './SubmitButton';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
test('Component "SubmitButton" is mount', () => {
  const component = renderer.create(
    <BrowserRouter>
      <SubmitButton  title={'Send'} disabled={false} onClick={()=>{}} />
    </BrowserRouter>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
