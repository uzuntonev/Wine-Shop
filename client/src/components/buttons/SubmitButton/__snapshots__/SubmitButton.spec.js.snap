// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component "SubmitButton" is mount 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-submit-1 MuiButton-containedPrimary MuiButton-fullWidth"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="submit"
>
  <span
    className="MuiButton-label"
  >
    Send
  </span>
</button>
`;
